File: 0004/P0004.cpp
#include <iostream>

bool isPalindrome(const std::string& s) {
  return s.size() == 6
      && s[0] == s[5]
      && s[1] == s[4]
      && s[2] == s[3]; 
}

int count(int n) {
  int result = 0;
  for (int i = 100; i < 1000; ++i) {
    for (int j = i; j < 1000; ++j) {
      int p = i * j;
      if (p >= n) break;
      std::string s = std::to_string(p);
      if (isPalindrome(s)) {
        result = std::max(result, p);
      }
    }
  }

  return result;
}

int main() {
  int t = 0;
  std::cin >> t;
  for (int i = 0; i < t; ++i) {
    int n = 0;
    std::cin >> n;
    int result = count(n);
    std::cout << result << std::endl;
  }
}

File: 0004/P0004.py
def reverse(n):
    s = str(n)
    return s[::-1]

def is_palindrome(n):
    s = str(n)
    rev = reverse(s)
    return rev == s

def has_six_digits(n):
    s = str(n)
    return len(s) == 6

def count(n):
    result = 0
    for i in range(100, 1000):
        for j in range(i, 1000):
            p = i * j
            if p >= n:
                break
            elif is_palindrome(p) and has_six_digits(p):
                result = max(result, p)
    return result

if __name__ == "__main__":
    t = int(input())
    for i in range(t):
        n = int(input())
        r = count(n)
        print(r)